mobile/lib/modules/home/ui/asset_grid/asset_grid_data_structure.dart

```diff
@@ -33,85 +35,112 @@ class RenderAssetGridElement {
   });
 }
 
-List<RenderAssetGridElement> assetsToRenderList(
-  List<Asset> assets,
-  int assetsPerRow,
-) {
-  List<RenderAssetGridElement> elements = [];
-
-  int cursor = 0;
-  while (cursor < assets.length) {
-    int rowElements = min(assets.length - cursor, assetsPerRow);
-    final date = assets[cursor].createdAt;
-
-    final rowElement = RenderAssetGridElement(
-      RenderAssetGridElementType.assetRow,
-      date: date,
-      assetRow: RenderAssetGridRow(
-        assets.sublist(cursor, cursor + rowElements),
-      ),
-    );
-
-    elements.add(rowElement);
-    cursor += rowElements;
-  }
-
-  return elements;
-}
-
-List<RenderAssetGridElement> assetGroupsToRenderList(
-  Map<String, List<Asset>> assetGroups,
-  int assetsPerRow,
-) {
-  List<RenderAssetGridElement> elements = [];
-  DateTime? lastDate;
-
-  assetGroups.forEach((groupName, assets) {
-    try {
-      final date = DateTime.parse(groupName);
-
-      if (lastDate == null || lastDate!.month != date.month) {
-        elements.add(
-          RenderAssetGridElement(
-            RenderAssetGridElementType.monthTitle,
-            title: groupName,
-            date: date,
-          ),
-        );
-      }
-
-      // Add group title
-      elements.add(
-        RenderAssetGridElement(
-          RenderAssetGridElementType.dayTitle,
-          title: groupName,
-          date: date,
-          relatedAssetList: assets,
-        ),
-      );
-
-      // Add rows
-      int cursor = 0;
-      while (cursor < assets.length) {
-        int rowElements = min(assets.length - cursor, assetsPerRow);
-
-        final rowElement = RenderAssetGridElement(
-          RenderAssetGridElementType.assetRow,
-          date: date,
-          assetRow: RenderAssetGridRow(
-            assets.sublist(cursor, cursor + rowElements),
-          ),
-        );
-
-        elements.add(rowElement);
-        cursor += rowElements;
-      }
-
-      lastDate = date;
-    } catch (e, stackTrace) {
-      log.severe(e, stackTrace);
+class RenderList {
+  final List<RenderAssetGridElement> elements;
+
+  RenderList(this.elements);
+
+  static Future<RenderList> fromAssetGroups(
+    Map<String, List<Asset>> assetGroups,
+    int assetsPerRow,
+  ) async {
+    const month_format_key = "month_format";
+    const day_format_key = "day_format";
+    const day_format_year_key = "day_format_year";
+    const groups_key = "groups";
+    const per_row_key = "per_row";
+
+    processData(Map<String, Object> data) {
```

Comment:
again, static class method (here especially because the nested method is super long)